plugins/modules/redfish_firmware_rollback.py

```diff
@@ -0,0 +1,345 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+#
+# Dell OpenManage Ansible Modules
+# Version 8.2.0
+# Copyright (C) 2023 Dell Inc. or its subsidiaries. All Rights Reserved.
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+#
+
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: redfish_firmware_rollback
+short_description: To perform a component firmware rollback using component name
+version_added: "8.2.0"
+description:
+  - This module allows the rollback of different components or versions of the iDRAC firmware available in iDRAC.
+  - Depending on the component, the firmware rollback is applied after an automatic or manual reboot.
+extends_documentation_fragment:
+  - dellemc.openmanage.redfish_auth_options
+options:
+  name:
+    type: str
+    required: true
+    description: Name or regular expression of the component to match and is case-sensitive.
+  reboot:
+    description:
+      - Reboot the server to apply the previous version of the firmware.
+      - C(true), reboots the server to rollback the firmware to available version.
+      - C(false) schedules the rollback of firmware until the next restart.
+      - When I(reboot) is C(false) some components will update immediately and the module will wait for
+        the server to come up.
+    type: bool
+    default: true
+  reboot_timeout:
+    type: int
+    description: Time in seconds to wait for the server to reboot.
+    default: 900
+requirements:
+  - "python >= 3.8.6"
+author:
+  - "Felix Stephen (@felixs88)"
+notes:
+  - Run this module from a system that has direct access to Redfish APIs.
```

Comment:
APIs or device/server/machine ?

Reply:
Documentation is still in review, will update this line as per the review